drivers/scheduler/k8s/specs/elasticsearch/px-elasticdata-app.yaml

```diff
@@ -112,9 +112,11 @@ spec:
   - metadata:
       name: es-data
       annotations:
-        volume.beta.kubernetes.io/storage-class: px-elasticdata-sc
         azure-storage-class: azure-es-sc
+        aws-storage-class: aws-sc-elasticsearch
```

Comment:
There is already support for different drivers by adding specific specs in driver directories. Why have 2 ways of doing this instead of using that?

Reply:
Support for that was added more than 3 months ago with this PR: https://github.com/portworx/torpedo/pull/382

Reply:
Well, I can spot a few problems with this approach for my case. I use portworx as a  packup driver, but in the same time volumes must be provisioned by particular cloud provider, but if I add cloud provisioner - backup won't work. Also implementing provisioner for cloud provider is in plans, but of out scope for this particular task. 
           I can see that only one provisioner from the map of provisioners is chosen, but for Azure and AWS I basically need  at least two - azure-disk and azure-file for example. But this simple mapping allows  to select only one provisioner to work with.
   So, taking this into account we decided with @harsh-px to resolve this by using annotations in PVC rather than selecting provisioners for storage classes.

Reply:
Not sure what you mean, we are using the same framework to test cloud provisions in the px-backup and stork tests.
Also, this is not changing the provisioner anywhere in the storageclass. Depending on the cloud to be tested it picks up the specs for the cloud from that directory. And you don't need to implement the cloud providers, they are already added for gcp, azure and aws.
If you want to have different storage classes you can add them to azure directory and use that. I don't see how you are doing that with the annotations method either.

Reply:
Okay, I put cloud specific spec to appropriate folder.But how I determine which storage class assign to each PVC?

Reply:
In the PVC you always use a storage class with a fixed name. In each of the cloud providers' directory, you create a storage class with that name and custom parameters. You can also create other cloud-specific specs in each directory if you want to create additional pvcs for some clouds.

Reply:
@disrani-px I've made a changes for that. Now we have generic specs in app folder and a bunch of provider specific subfolders with specs. So we will never create redunant entities.